### PR TITLE
refactor: remove getComposedData from all charts that are on selectors

### PR DESCRIPTION
**Pull request by ckifer**
## Description

- removes all removable `getComposedData` references
- breaks `Customized`

## Related Issue

https://github.com/recharts/recharts/issues/4551

## Motivation and Context

Keep...moving...

## How Has This Been Tested?

- npm test
- customized breaks in storybook
- no chart diff

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a storybook story or extended an existing story to show my changes

### CODE REVIEW

File: test/cartesian/Area.spec.tsx [hunk 1]
```diff
@@ -892,262 +892,3 @@ describe('getBaseValue', () => {
     });
   });
 });
-
-describe('getComposedData', () => {
```
Comment (PavelVanecek) [inline test/cartesian/Area.spec.tsx:896]: The function and the logic is still there. Not in a static method but in a function. Can we keep the tests?

File: test/chart/BarChart.spec.tsx [hunk 2]
```diff
@@ -599,27 +599,15 @@ describe('<BarChart />', () => {
     });
 
     test('Stacked bars are actually stacked', () => {
-      let seriesOneBarOneValue, seriesTwoBarOneValue;
-      const Spy = (props: { formattedGraphicalItems?: any }) => {
```
Comment (PavelVanecek) [inline test/chart/BarChart.spec.tsx:603]: Can we replace this with selectors?

File: test/state/selectors/scatterSelectors.spec.tsx [hunk 3]
```diff
@@ -33,23 +33,26 @@ describe('selectScatterPoints', () => {
...
-    render(
-      <PieChart width={100} height={200}>
-        <Pie data={pageData} dataKey="uv" />
-        <Customized component={<Comp />} />
-      </PieChart>,
-    );
-    expect(scatterPointsSpy).toHaveBeenCalledWith(undefined);
-    expect(scatterPointsSpy).toHaveBeenCalledTimes(2);
-  });
+  // TODO: unsure why this does not work... does PieChart prevent rendering for some reason?
```
Comment (PavelVanecek) [inline test/state/selectors/scatterSelectors.spec.tsx:36]: What happens here? The selector should continue returning undefined

File: test/cartesian/Area.spec.tsx (hunk 1 above)
Comment (ckifer) [reply test/cartesian/Area.spec.tsx:896]: Yeah I can do the same for the compute function I guess

File: test/state/selectors/scatterSelectors.spec.tsx (hunk 3 above)
Comment (ckifer) [reply test/state/selectors/scatterSelectors.spec.tsx:36]: The selector is never called and I don't know why

File: test/chart/BarChart.spec.tsx (hunk 2 above)
Comment (ckifer) [reply test/chart/BarChart.spec.tsx:603]: Maybe but I've been having trouble with the selectors in tests

Comment (ckifer) [reply test/chart/BarChart.spec.tsx:603]: done

File: test/state/selectors/scatterSelectors.spec.tsx (hunk 3 above)
Comment (ckifer) [reply test/state/selectors/scatterSelectors.spec.tsx]: I added `it.fails` and we can revisit. Nothing is rendering children in Pie specifically

We should allow children to render children at the chart level I guess..

**Comment by ckifer** [comment]
Added 2 of 3 groups of tests back, going to merge this

**Pull request merged**